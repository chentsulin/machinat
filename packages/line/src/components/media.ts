import { makeUnitSegment } from '@machinat/core/renderer';
import { UnitSegment, FunctionOf } from '@machinat/core/renderer/types';
import { annotateLineComponent } from '../utils';
import { LineComponent, LineMessageSegmentValue } from '../types';

/**
 * @category Props
 */
export type VideoProps = {
  /** URL of video file (Max character limit: 1000) */
  originalContentUrl: string;
  /** URL of preview image (Max character limit: 1000) */
  previewImageUrl: string;
  /**
   * ID used to identify the video when Video viewing complete event occurs. If
   * you send a video message with trackingId added, the video viewing complete
   * event occurs when the user finishes watching the video.
   */
  trackingId?: string;
};

/** @internal */
const __Video: FunctionOf<LineComponent<
  VideoProps,
  UnitSegment<LineMessageSegmentValue>
>> = function Video(node, path) {
  const { originalContentUrl, previewImageUrl, trackingId } = node.props;

  return [
    makeUnitSegment(node, path, {
      type: 'video' as const,
      originalContentUrl,
      previewImageUrl,
      trackingId,
    }),
  ];
};
/**
 * Video sends an video message.
 * @category Component
 * @props {@link VideoProps}
 * @guides Check official [reference](https://developers.line.biz/en/reference/messaging-api/#video-message).
 */
export const Video: LineComponent<
  VideoProps,
  UnitSegment<LineMessageSegmentValue>
> = annotateLineComponent(__Video);

/**
 * @category Props
 */
export type AudioProps = {
  /** URL of audio file (Max character limit: 1000) */
  originalContentUrl: string;
  /** Length of audio file (milliseconds) */
  duration: number;
};

/** @internal */
const __Audio: FunctionOf<LineComponent<
  AudioProps,
  UnitSegment<LineMessageSegmentValue>
>> = function Audio(node, path) {
  const { originalContentUrl, duration } = node.props;
  return [
    makeUnitSegment(node, path, {
      type: 'audio' as const,
      originalContentUrl,
      duration,
    }),
  ];
};
/**
 * Audio sends an audio message.
 * @category Component
 * @props {@link AudioProps}
 * @guides Check official [reference](https://developers.line.biz/en/reference/messaging-api/#video-message).
 */
export const Audio: LineComponent<
  AudioProps,
  UnitSegment<LineMessageSegmentValue>
> = annotateLineComponent(__Audio);
