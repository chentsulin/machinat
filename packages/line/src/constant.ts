export const CHANNEL_REQUEST_GETTER = Symbol(
  'channel_request_getter.line.machinat'
);
export const BULK_REQUEST_GETTER = Symbol('bulk_request_getter.line.machinat');

/** @internal */
export const LINE = 'line' as const;

/** @internal */
export const PATH_REPLY = 'v2/bot/message/reply';

/** @internal */
export const PATH_PUSH = 'v2/bot/message/push';

/** @internal */
export const PATH_MULTICAST = 'v2/bot/message/multicast';

/** @internal */
export const PATH_RICHMENU = 'v2/bot/richmenu';

export enum LiffContextOs {
  Ios,
  Android,
  Web,
}
