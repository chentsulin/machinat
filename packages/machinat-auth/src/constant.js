// @flow
import { namedInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const AUTH_SERVER_AUTHORIZERS_I = namedInterface<
  ServerAuthorizer<any, any>[]
>({
  name: 'ServerAuthorizersList',
  multi: true,
});

export const AUTH_MODULE_CONFIGS_I = namedInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
