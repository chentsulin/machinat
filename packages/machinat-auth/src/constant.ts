import { makeInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const SERVER_AUTHORIZERS_I = makeInterface<ServerAuthorizer<any, any>[]>(
  { name: 'ServerAuthorizersList', multi: true }
);

export const MODULE_CONFIGS_I = makeInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
