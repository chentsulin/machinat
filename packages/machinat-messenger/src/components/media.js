import { unitSegment } from '@machinat/core/renderer';
import { annotateMessengerComponent } from '../utils';

const mediaFactory = (name, type) => {
  const container = {
    [name]: (node, path) => {
      const { url, reusable, attachmentId } = node.props;

      return [
        unitSegment(node, path, {
          message: {
            attachment: {
              type,
              payload: {
                url,
                is_reusable: reusable,
                attachment_id: attachmentId,
              },
            },
          },
        }),
      ];
    },
  };

  return annotateMessengerComponent(container[name]);
};

export const Image = mediaFactory('Image', 'image');
export const Video = mediaFactory('Video', 'video');
export const Audio = mediaFactory('Audio', 'audio');
export const File = mediaFactory('File', 'file');
