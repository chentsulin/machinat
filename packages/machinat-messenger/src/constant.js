// @flow
export const MESSENGER = 'messenger';
export const MESSENGER_NAMESPACE = 'Messenger';
export const MESSENGER_NATIVE_TYPE = Symbol.for('machinat.messenger.native');

export const ATTACHED_FILE_DATA = Symbol(
  'machinat.messenger.job.attached_file_data'
);

export const ATTACHED_FILE_INFO = Symbol(
  'machinat.messenger.job.attached_file_info'
);

export const ENTRY_MESSAGES = 'me/messages';

export const ENTRY_PASS_THREAD_CONTROL = 'me/pass_thread_control';
export const ENTRY_REQUEST_THREAD_CONTROL = 'me/request_thread_control';
export const ENTRY_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const ENTRY_MESSAGE_CREATIVES = 'me/message_creatives';
export const ENTRY_BROADCAST_MESSAGES = 'me/broadcast_messages';
export const ENTRY_CUSTOM_LABELS = 'me/custom_labels';
