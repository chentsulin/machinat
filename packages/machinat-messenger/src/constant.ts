export const MESSENGER = 'messenger';

export const ATTACHMENT_DATA = Symbol('messenger.segment.attachment_data');

export const ATTACHMENT_INFO = Symbol('messenger.segment.attachment_info');

export const ASSET_TAG = Symbol.for('messenger.segment.asset_tag');
export const ENTRY_PATH = Symbol.for('messenger.segment.entry');

export const PATH_MESSAGES = 'me/messages';

export const PATH_PASS_THREAD_CONTROL = 'me/pass_thread_control';
export const PATH_REQUEST_THREAD_CONTROL = 'me/request_thread_control';
export const PATH_TAKE_THREAD_CONTROL = 'me/take_thread_control';

export const PATH_MESSAGE_ATTACHMENTS = 'me/message_attachments';
export const PATH_MESSENGER_PROFILE = 'me/messenger_profile';
export const PATH_PERSONAS = 'me/personas';
