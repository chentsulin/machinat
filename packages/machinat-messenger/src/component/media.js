import { ATTACHED_FILE_DATA, ATTACHED_FILE_INFO } from '../symbol';

import { renderOnlyInTypes, annotateNativeRoot, getRendered } from './utils';
import { ENTRY_MESSAGES } from './constant';
import * as quickRepliesComponents from './quickReply';

const quickReplyTypes = Object.values(quickRepliesComponents);

const nativeMediaFactroy = (name, type) => {
  const container = {
    [name]: (
      { url, quickReplies, metadata, reusable, attachmentId, data, fileInfo },
      render
    ) => {
      const repliesResult = renderOnlyInTypes(
        quickReplyTypes,
        quickReplies,
        render,
        'quickReplies'
      );

      return {
        message: {
          quick_replies: repliesResult && repliesResult.map(getRendered),
          metadata,
          attachment: {
            type,
            payload: {
              url,
              is_reusable: reusable,
              attachment_id: attachmentId,
            },
          },
        },
        [ATTACHED_FILE_DATA]: data,
        [ATTACHED_FILE_INFO]: fileInfo,
      };
    },
  };

  return annotateNativeRoot(container[name], ENTRY_MESSAGES);
};

export const Image = nativeMediaFactroy('Image', 'image');
export const Video = nativeMediaFactroy('Video', 'video');
export const Audio = nativeMediaFactroy('Audio', 'audio');
export const File = nativeMediaFactroy('File', 'file');
