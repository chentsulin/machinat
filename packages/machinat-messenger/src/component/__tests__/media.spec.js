import Machinat from 'machinat';

import { MESSENGER_NAITVE_TYPE } from '../../constant';
import { Image, Video, Audio, File } from '../media';
import renderHelper from './renderHelper';

const renderInside = jest.fn();
const render = renderHelper(renderInside);

describe('media Components', () => {
  test.each([Image, Video, Audio, File])('is valid root Component', Media => {
    expect(typeof Media).toBe('function');
    expect(Media.$$native).toBe(MESSENGER_NAITVE_TYPE);
    expect(Media.$$entry).toBe('me/messages');
    expect(Media.$$namespace).toBe('Messenger');
  });

  it('match snapshot', () => {
    expect(
      [
        <Image url="http://this.is/a/picture" reusable />,
        <Video url="http://this.is/a/video" reusable />,
        <Audio url="http://this.is/an/audio" reusable />,
        <File url="http://this.is/a/file" reusable />,
        <Image attachmentId="_I_am_Image_" />,
        <Video attachmentId="_I_am_Video_" />,
        <Audio attachmentId="_I_am_Audio_" />,
        <File attachmentId="_I_am_File_" />,
      ].map(render)
    ).toMatchSnapshot();
  });
});
