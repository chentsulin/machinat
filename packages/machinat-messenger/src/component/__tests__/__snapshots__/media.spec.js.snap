// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media Components match snapshot 1`] = `
Array [
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "reusable": true,
          "url": "http://this.is/a/picture",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": undefined,
              "is_reusable": true,
              "url": "http://this.is/a/picture",
            },
            "type": "image",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "reusable": true,
          "url": "http://this.is/a/video",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": undefined,
              "is_reusable": true,
              "url": "http://this.is/a/video",
            },
            "type": "video",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "reusable": true,
          "url": "http://this.is/an/audio",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": undefined,
              "is_reusable": true,
              "url": "http://this.is/an/audio",
            },
            "type": "audio",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "reusable": true,
          "url": "http://this.is/a/file",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": undefined,
              "is_reusable": true,
              "url": "http://this.is/a/file",
            },
            "type": "file",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "attachmentId": "_I_am_Image_",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": "_I_am_Image_",
              "is_reusable": undefined,
              "url": undefined,
            },
            "type": "image",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "attachmentId": "_I_am_Video_",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": "_I_am_Video_",
              "is_reusable": undefined,
              "url": undefined,
            },
            "type": "video",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "attachmentId": "_I_am_Audio_",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": "_I_am_Audio_",
              "is_reusable": undefined,
              "url": undefined,
            },
            "type": "audio",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
  Array [
    Object {
      "node": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "attachmentId": "_I_am_File_",
        },
        "type": [Function],
      },
      "path": "$",
      "type": "unit",
      "value": Object {
        "message": Object {
          "attachment": Object {
            "payload": Object {
              "attachment_id": "_I_am_File_",
              "is_reusable": undefined,
              "url": undefined,
            },
            "type": "file",
          },
        },
        Symbol(machinat.messenger.job.attached_file_data): undefined,
        Symbol(machinat.messenger.job.attached_file_info): undefined,
      },
    },
  ],
]
`;
